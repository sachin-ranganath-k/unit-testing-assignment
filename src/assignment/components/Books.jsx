import React from "react";

function Books(propsFromBookList) {
  return (
    <div>
      <h2>Welcome to My Favorite Book Portal</h2>
      <table border="2">
        <tbody>
          <tr>
            <th>Book Name</th>
            <th>Book Author</th>
          </tr>
          <tr>
            <td data-testid="book-name">{propsFromBookList.bookName}</td>
            <td data-testid="book-author">{propsFromBookList.bookAuthor}</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}

export default Books;
