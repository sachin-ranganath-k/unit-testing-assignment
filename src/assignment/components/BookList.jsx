import React, { Component } from "react";
import Books from "./Books";

class BookList extends Component {
  render() {
    const bookName = "Web Programming";
    const bookAuthor = "Stalin";

    return (
      <div>
        <Books bookName={bookName} bookAuthor={bookAuthor} />
      </div>
    );
  }
}

export default BookList;
