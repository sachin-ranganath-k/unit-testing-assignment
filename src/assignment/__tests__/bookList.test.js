import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import BookList from "../components/BookList";
import Books from "../components/Books";

describe("BookList component", () => {
  it("should render BookList component", () => {
    render(<BookList />);
  });

  it("receive props from BookList component", () => {
    render(<Books bookName="Web Programming" bookAuthor="Stalin" />);
    expect(screen.getByTestId("book-name")).toHaveTextContent("Web Programming");
    expect(screen.getByTestId("book-author")).toHaveTextContent("Stalin");
  });

  it("snapshot testing", () => {
    const snapshot = renderer.create(
      <Books bookName="Web Programming" bookAuthor="Stalin" />
    );
    expect(snapshot).toMatchSnapshot();
  });

  it("should check for a text", () => {
    render(<Books />);
    const text = screen.getByText("Welcome to My Favorite Book Portal");
    expect(text).toBeInTheDocument();
  });
});
