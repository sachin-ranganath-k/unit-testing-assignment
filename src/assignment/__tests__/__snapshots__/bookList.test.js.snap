// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component snapshot testing 1`] = `
<div>
  <h2>
    Welcome to My Favorite Book Portal
  </h2>
  <table
    border="2"
  >
    <tbody>
      <tr>
        <th>
          Book Name
        </th>
        <th>
          Book Author
        </th>
      </tr>
      <tr>
        <td
          data-testid="book-name"
        >
          Web Programming
        </td>
        <td
          data-testid="book-author"
        >
          Stalin
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
